### ISSUE TITLE
Drone timed out in nightly tests during vic-machine create (waiting for IP information)

### ISSUE THREAD

**Issue by chengwang86** [vmware/vic#5447]
This is seen in 13-1-vMotion-VCH.

[13-1-vMotion-VCH-Appliance.zip](https://github.com/vmware/vic/files/1069278/13-1-vMotion-VCH-Appliance.zip)

An error occurred at [13-1-vMotion-VCH.robot](https://github.com/vmware/vic/blob/master/tests/manual-test-cases/Group13-vMotion/13-1-vMotion-VCH-Appliance.robot#L68) --> [cleanup dangling containers on test server](https://github.com/vmware/vic/blob/master/tests/resources/VCH-Util.robot#L349) --> [Get scratch disk from vm info](https://github.com/vmware/vic/blob/master/tests/resources/VCH-Util.robot#L346) 

However, the same error occurs in every successful run and is always ignored. So it's not clear why the test exits after this error.

I have checked the timestamps when the test started and exited. The time difference is less than one hour so I don't think this is due to nimbus timeout.

**Comment by chengwang86** [comment]
Also seen here on vsphere 6.0
[5-6-1-VSAN-Simple.zip](https://github.com/vmware/vic/files/1072492/5-6-1-VSAN-Simple.zip)

**Comment by chengwang86** [comment]
Update:

Till the test code where the test exited, there were no test errors. The test exited during `vic-machine create`:
```
Jun 12 2017 17:38:33.234Z INFO  Uploading images for container
Jun 12 2017 17:38:33.234Z INFO  	"bin/bootstrap.iso"
Jun 12 2017 17:38:33.234Z DEBUG target delete path = bcd13e59-0553-9555-29a8-02001b0368f5/V1.1.0-RC3-11382-4120DE4-bootstrap.iso
Jun 12 2017 17:38:33.246Z INFO  	"bin/appliance.iso"
Jun 12 2017 17:38:33.246Z DEBUG target delete path = bcd13e59-0553-9555-29a8-02001b0368f5/V1.1.0-RC3-11382-4120DE4-appliance.iso
Jun 12 2017 17:38:40.297Z DEBUG Will not try again. Operation succeeded
Jun 12 2017 17:38:52.888Z DEBUG Will not try again. Operation succeeded
Jun 12 2017 17:48:57.400Z INFO  Waiting for IP information
```

**Comment by chengwang86** [comment]
Update:

From the server where we run the nightly tests, I checked this file `/home/vicadmin/nightly-log/nightly_06-11-2017.txt` and found the following
```
Manual-Test-Cases.Group13-vMotion.13-1-vMotion-VCH-Appliance :: Test 13-1 -...
==============================================================================

Starting basic VSAN cluster deploy...
Set environment variables up for GOVC
Create a distributed switch
Create three new distributed switch port groups for management and vm network traffic
Add all the hosts to the distributed switch
Enable DRS and VSAN on the cluster
Deploy VIC to the VC cluster
Step 6-9
TEST_URL=10.160.161.171

Installing VCH to test server...
Installer completed successfully: VCH-0-9042...
| PASS |
------------------------------------------------------------------------------
Step 10-13
TEST_URL=10.160.161.171
Destroying dangling VCH: VCH-0-9042

Installing VCH to test server...
[info] Timeout request received, killing process
[DRONE] finished job #1
[DRONE] exit code 128
[DRONE] job #1 failed
[DRONE] build failed
Failed
```

**Comment by chengwang86** [comment]
Regarding to `timeout for waiting for IP information`:

```
// ensureApplianceInitializes checks if the appliance component processes are launched correctly
func (d *Dispatcher) ensureApplianceInitializes(conf *config.VirtualContainerHostConfigSpec) error {
	defer trace.End(trace.Begin(""))

	if d.appliance == nil {
		return errors.New("cannot validate appliance due to missing VM reference")
	}

	log.Infof("Waiting for IP information")
	d.waitForKey(extraconfig.CalculateKeys(conf, "ExecutorConfig.Networks.client.Assigned.IP", "")[0])
	ctxerr := d.ctx.Err()

	if ctxerr == nil {
		log.Info("Waiting for major appliance components to launch")
		for _, k := range extraconfig.CalculateKeys(conf, "ExecutorConfig.Sessions.*.Started", "") {
			d.waitForKey(k)
		}
	}
```
If `d.waitForKey(extraconfig.CalculateKeys...` takes longer than the timeout value in the context, vic-machine create should fail with error msgs.

**Comment by chengwang86** [comment]
According to this post https://stackoverflow.com/a/41366962 which points to drone's source code, we can set both `DRONE_TIMEOUT` and `DRONE_TIMEOUT_INACTIVITY`.

**Comment by chengwang86** [comment]
Update:

According to @mhagen-vmware , currently our drone version on the nightly test server is 0.4 which does not allow us to customize `DRONE_TIMEOUT` and the hardcoded value is 60min.

Based on discussion with @emlin , it could be the case that `vic-machine create` hangs for 30min during waiting for `ExecutorConfig.Networks.client.Assigned.IP`. This relates to the DHCP server in the test environment. However, since the test environment is cleaned up after each run, we cannot further investigate the reason why it took so long to obtain an IP.

Possible solutions:
- upgrade drone to version 0.5, which we use for the ci test, and customize the `DRONE_TIMEOUT` to a higher value.
- decrease the `TEST_TIMEOUT` of `vic-machine create`

**Comment by chengwang86** [comment]
Based on my discussion with @mhagen-vmware , we decide to set the `TEST_TIMEOUT` for `vic-machine create` to 15min.

**Comment by chengwang86** [comment]
fixed in PR #5869
